src/task/index_task.rs

```diff
@@ -0,0 +1,109 @@
+use crate::cluster::{Cluster};
+use std::sync::Arc;
+use crate::errors::{ErrorKind, Result};
+use std::time::{Duration};
+use crate::task::{Status, Task};
+
+
+/// Struct for querying index creation status
+#[derive(Debug, Clone)]
+pub struct IndexTask {
+    cluster: Arc<Cluster>,
+    namespace: String,
+    index_name: String
+}
+
+
+static SUCCESS_PATTERN: &'static str = "load_pct=";
+static FAIL_PATTERN_201: &'static str = "FAIL:201";
+static FAIL_PATTERN_203: &'static str = "FAIL:203";
+static DELMITER: &'static str = ";";
+
+impl IndexTask {
+    /// Initializes IndexTask from client, creation should only be expose to Client
+    pub fn new(cluster: Arc<Cluster>, namespace: String, index_name: String) -> Self {
+        IndexTask {
+        	cluster: cluster,
+        	namespace: namespace,
+            index_name: index_name
+        }
+    }
+
+    fn build_command(namespace: String, index_name: String) -> String {
+        return format!("{}{}{}{}", "sindex/", namespace, "/", index_name);
+    }
+
+	fn parse_response(response: &str) -> Result<Status> {
+        match response.find(SUCCESS_PATTERN) {
+            None => {
+                match (response.find(FAIL_PATTERN_201), response.find(FAIL_PATTERN_203)) {
+                    (None, None) => bail!(ErrorKind::BadResponse(format!("Code 201 and 203 missing. Response: {}", response))),
+                    (_, _) => return Ok(Status::NotFound)
+                }
+            },
+            Some(pattern_index) => {
+                let percent_begin = pattern_index + SUCCESS_PATTERN.len();
+
+                let percent_end = match response[percent_begin..].find(DELMITER) {
+                    None =>  bail!(ErrorKind::BadResponse(format!("delimiter missing in response. Response: {}", response))),
+                    Some(percent_end) => percent_end
+                };
+                let percent_str = &response[percent_begin..percent_begin+percent_end];
+                if percent_str.parse::<isize>().unwrap() != 100 {
+                    return Ok(Status::InProgress);
+                } else {
+                    return Ok(Status::Complete);
+                }
```

Comment:
Suggestion to improve the error handling:

```suggestion
                match percent_str.parse::<isize>() {
                    Ok(100) => return Ok(Status::Complete),
                    Ok(_) => return Ok(Status::InProgress),
                    Err(_) => bail!(ErrorKind::BadResponse(
                        "Unexpected load_pct value from server".to_string()
                    )),
                }
```